src/components/tables/DataExplorerTable/index.jsx

```diff
@@ -1,17 +1,133 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import FileSaver from 'file-saver';
 import { Table } from '@arranger/components/dist/Arranger';
+import dictIcons from '../../../img/icons/index';
+import IconicButton from '../../buttons/IconicButton';
 import './DataExplorerTable.less';
 
+const DOWNLOAD_MANIFEST_BUTTON_TEXT = 'Download Manifest';
+const UUID_FIELD = 'uuid';
+const SUBJECT_ID_FIELD = 'subject_id';
+const FILE_TYPE_NAME = 'file';
+const FILENAME = 'manifest.json';
+const GRAPHQL_ENDPOINT = '/graphql';
+const MSG_FAILED_DOWNLOAD = 'Failed to download manifest file';
+
+const constructGraphQLQuery = (selectedTableRows, isGettingCount, count) => {
+  const sqonObj = {
+    op: 'and',
+    content: [
+      {
+        op: 'in',
+        content: {
+          field: SUBJECT_ID_FIELD,
+          value: [...selectedTableRows],
+        },
+      },
+    ],
+  };
+  const getContentQuery = `edges {
+                          node {
+                            ${UUID_FIELD}
+                            ${SUBJECT_ID_FIELD}
+                          }
+                        }`;
+  const getCountQuery = 'total';
+  const gqlQuery = {
+    query: `query ($first: Int, $sqon: JSON){
+                ${FILE_TYPE_NAME} {
+                  hits (first: $first, filters: $sqon) {
+                    ${isGettingCount ? getCountQuery : getContentQuery}
+                  }
+                }
+              }`,
+    variables: {
+      sqon: sqonObj,
+      first: isGettingCount ? 0 : count,
+    },
+  };
+  return gqlQuery;
+};
+
+const queryFileMappingData = async (apiFunc, projectId, selectedTableRows) => {
+  const countQuery = await apiFunc({
+    endpoint: '/'.concat(projectId, GRAPHQL_ENDPOINT),
+    body: constructGraphQLQuery(selectedTableRows, true),
+  });
+  if (!countQuery) {
+    throw MSG_FAILED_DOWNLOAD;
+  }
+  const count = countQuery.data[FILE_TYPE_NAME].hits.total;
+  const manifest = await apiFunc({
+    endpoint: '/'.concat(projectId, GRAPHQL_ENDPOINT),
+    body: constructGraphQLQuery(selectedTableRows, false, count),
+  });
+  if (!manifest) {
+    throw MSG_FAILED_DOWNLOAD;
+  }
+  const manifestJSON = manifest.data[FILE_TYPE_NAME].hits.edges.map(item => item.node);
+  return manifestJSON;
+};
+
+const replaceToolbarText = (selectedTableRows) => {
+  const arrangerTableClassName = 'tableToolbar';
+  const toolbarTextElement = document.querySelector(`.data-explorer-table__arranger-table div.${arrangerTableClassName} div:first-child`);
+  if (toolbarTextElement) {
```

Comment:
Why are you doing this?  We shouldn't have to interact directly with the DOM with ReactJS ...

Reply:
This is because `arranger table` always has a fixed text `1-20 of XXX subjects` in  `TableToolBar`, and donnot allow us to change text, and our design need to change this text...Or do you think we ask arranger team to make this customizable

Reply:
I would push for not changing this text

Reply:
Great, then it would be easier to just keep the text no changed...